### ISSUE TITLE
Documentation Idea

### ISSUE THREAD

**Issue by black7375** [NICHOLAS85/z-a-eval#2]
**[Zinit Wiki](https://zdharma.org/zinit/wiki/)**

I think this is useful and worth mentioning in the official document.
- https://github.com/zdharma/zinit/tree/documentation
- https://github.com/zdharma/zinit/tree/gh-pages

**Before/After**
It's intuitive what kind of situation you need.

**Null Plugin Eval**
It is good to have an example when caching non-plugin things. (like with `id-as`)
- https://github.com/zdharma/zinit/issues/68

**Comment by NICHOLAS85** [comment]
Sure I could improve the documentation to include some Before/After invocations and Null Plugin examples. Feel free to share some examples here and I'll look into including them.

**Comment by black7375** [comment]
**Before/After**
- Simple
[zoxide](https://github.com/ajeetdsouza/zoxide#zsh) is a good example.
It is simple and because README states to use `eval`.
```shell
## Before
zinit ice as"command" from"gh-r" mv"zoxide* -> zoxide"  \
      atclone"./zoxide init zsh > init.zsh"  atpull"%atclone" src"init.zsh" nocompile'!'
zinit light ajeetdsouza/zoxide

## After
zinit ice as"command" from"gh-r" mv"zoxide* -> zoxide" \
      eval"./zoxide init zsh"
zinit light ajeetdsouza/zoxide```
```

- Others
It's a bit more complicated case.
```shell
## Before
# trapd00r/LSCOLORS: https://zdharma.org/zinit/wiki/LS_COLORS-explanation/
zinit ice atclone"dircolors -b LS_COLORS > init.zsh" \
      atpull"%atclone" pick"init.zsh" nocompile'!' \
      atload'zstyle ":completion:*" list-colors “${(s.:.)LS_COLORS}”'
zinit light trapd00r/LS_COLORS

# dbrgn/tealdeer
_tldr-setting() {
    tldr --update
    local comp="$(tldr 2>/dev/null --list)"
    echo "complete -W \"$comp\" tldr" > init.zsh
}
zinit ice as"command" from"gh-r" \
      mv"tldr* -> tldr" atclone"_tldr-setting" atpull"%atclone" pick"init.zsh"
zinit light dbrgn/tealdeer


## After
# trapd00r/LSCOLORS
zinit ice wait"2" eval"dircolors -b LS_COLORS" \
      atload'zstyle ":completion:*" list-colors “${(s.:.)LS_COLORS}”'
zinit light trapd00r/LS_COLORS

# dbrgn/tealdeer
zinit ice wait"2" as"command" from"gh-r" \
      mv"tldr* -> tldr" atclone"./tldr --update" atpull"%atclone" \
      eval'echo "complete -W \"$(./tldr 2>/dev/null --list)\" tldr"'
zinit light dbrgn/tealdeer
```

**Null Plugin Eval**
You can use it to cache `eval` results.
Use with a `id-as`.
```shell
## Before
# thefuck -- simple example
eval $(thefuck --alias)

# kubectl -- complex example
if [[ "${+commands[kubectl]}" == 1 ]]; then
    eval $(kubectl completion zsh)
fi

## After
# thefuck
zinit ice id-as"thefuck_alias" eval"thefuck --alias"
zinit light zdharma/null

# kubectl
zinit ice id-as"kubectl_completion" has"kubectl" \
      eval"kubectl completion zsh"
zinit light zdharma/null
```


Will this be enough?
If README becomes complicated, it is also good to link this issue.

**Comment by NICHOLAS85** [comment]
Alright included some examples and linked this issue as a reference for more!
